docs/app-gen.md

```diff
@@ -0,0 +1,118 @@
+      --name="": Set name to use for generated application artifacts
+      --output-image="": Image name to use for output
+  -p, --ports="": Comma-separated list of ports to expose on pod deployment
+      --ref="": Set the name of the repository branch/ref to use
+      --source-url="": Set the source URL
+      --token="": If present, the bearer token for this request.
+```
+
+## Output
+The output of app-gen is a Kubernetes ```List``` resource in json format that contains at least the following:
+
+* BuildConfig - a build configuration that builds the specified source repository
+* ImageRepository - an image repository that points to the image built by the BuildConfig
+* DeploymentConfig - a deployment config that listens to new images pushed to the image repository and deploys a pod with that image
+* Service(s) - services that expose the ports exposed by the image from the built source.
+
+The output of app-gen can be directly applied to an OpenShift server by using the create command. Following is an example flow for using app-gen
+
+```sh
+# go to the directory that contains the application source
+cd ruby-hello-world
+# generate the application objects and create them on the openshift server
+openshift ex app-gen | osc create -f -
+# start a new build using the newly-created build configuration
+osc start-build ruby-hello-world
+```
+
+
+## Source Repository
+The input of app-gen is a source repository. This may be a local directory or a repository specified as a URL.
+
+### Local directory
+If app-gen is executed with no parameter, the current working directory is used. Otherwise, the directory passed in will be used. The repository must be a git repository, however, the directory does not have to be the root of the git repository.
+
+If app-gen determines that the directory is a git repository, it will determine the following using git:
+
+* The root directory of the repository
+* The remote URL fo the git repository. For this, it will use the ```origin``` remote. OpenShift must be able to download the source from this URL in order to build the code. It does not use the content of the source directory directly. This has the following implications:
+  * The repository URL must be accessible from the OpenShift node executing the builds.
+  * The repository must be publicly accessible. This means that git repositories using the SSH protocol (for example: ```git@github.com:openshift/ruby-hello-world.git```) are not yet supported.
+  * Any code that is only available locally and not in the URL-accessible repository will not be picked up for the build.
+* The currently checkedout branch name. It is assumed that the branch exists in the remote repository as well. If a different branch needs to be used, specify the *--ref* flag in the command line.
+
+### Repository URL
+If a repository URL is specified, app-gen will use that URL to create the buildConfig. The URL may include a specific branch to use by appending a hash with the branch name:
```

Comment:
does this # syntax allow the use of any git ref, not just branches?  I assume it does...